gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatchResult.java

```diff
@@ -0,0 +1,48 @@
+package com.google.gcloud.dns;
+
+import com.google.gcloud.BatchResult;
+
+/**
+ * This class holds a single result of a batch call to the Cloud DNS.
+ */
+public class DnsBatchResult<T> implements BatchResult<T, DnsException> {
+
+  private T result;
+  private boolean submitted = false;
+  private DnsException error;
+
+  DnsBatchResult() {
+  }
+
+  @Override
+  public boolean submitted() {
+    return submitted;
+  }
+
+  @Override
+  public T get() throws DnsException {
+    if(!submitted()) {
```

Comment:
I think this could be done in the `BatchResult` which maybe can be changed to be an abstract class instead of an interface. If you have a protected method that set result or an error, sub-classes could call it.